lisa/sut_orchestrator/aws/features.py

```diff
@@ -0,0 +1,503 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import base64
+from dataclasses import dataclass
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Type
+
+from assertpy import assert_that
+from dataclasses_json import dataclass_json
+
+from lisa import features, schema, search_space
+from lisa.features.gpu import ComputeSDK
+from lisa.node import Node
+from lisa.operating_system import CentOs, Redhat, Suse, Ubuntu
+from lisa.util import LisaException
+
+if TYPE_CHECKING:
+    from .platform_ import AwsPlatform
+
+from .. import AWS
+from .common import AwsNodeSchema, get_node_context
+
+
+class AwsFeatureMixin:
+    def _initialize_information(self, node: Node) -> None:
+        node_context = get_node_context(node)
+        self._intsance_id = node_context.intsance_id
+
+
+class StartStop(AwsFeatureMixin, features.StartStop):
+    def _stop(self, wait: bool = True) -> None:
+        platform: AwsPlatform = self._platform  # type: ignore
+        ec2_client = platform._ec2_client
+        ec2_client.stop_instances(InstanceIds=[self._intsance_id])
+        ec2_client.get_waiter("instance_stopped").wait()
+
+    def _start(self, wait: bool = True) -> None:
+        platform: AwsPlatform = self._platform  # type: ignore
+        ec2_client = platform._ec2_client
+        ec2_client.start_instances(InstanceIds=[self._intsance_id])
+        ec2_client.get_waiter("instance_running").wait()
+
+    def _restart(self, wait: bool = True) -> None:
+        platform: AwsPlatform = self._platform  # type: ignore
+        ec2_client = platform._ec2_client
+        ec2_client.reboot_instances(InstanceIds=[self._intsance_id])
+        ec2_client.get_waiter("instance_running").wait()
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        super()._initialize(*args, **kwargs)
+        self._initialize_information(self._node)
+
+
+def get_aws_disk_type(disk_type: schema.DiskType) -> str:
+    assert isinstance(disk_type, schema.DiskType), (
+        f"the disk_type must be one value when calling get_disk_type. "
+        f"But it's {disk_type}"
+    )
+
+    result = _disk_type_mapping.get(disk_type, None)
+    assert result, f"unkonwn disk type: {disk_type}"
+
+    return result
+
+
+_disk_type_mapping: Dict[schema.DiskType, str] = {
+    schema.DiskType.PremiumSSDLRS: "io1",
+    schema.DiskType.StandardHDDLRS: "st1",
+    schema.DiskType.StandardSSDLRS: "gp3",
+}
+
+# disk types are ordered by commonly and cost. The earlier is lower cost.
+_ordered_disk_types: List[schema.DiskType] = [
```

Comment:
If some items are shared between Azure and Aws. We can create a common.py under sut_orchestrator to put them.

Reply:
When rebase to latest change on main branch, these definition is moved to the common schema.py.